MAUI/Scheduler/load-on-demand.md

```diff
@@ -0,0 +1,311 @@
+{% highlight xaml %}
+
+ <scheduler:SfScheduler x:Name="Scheduler" 
+                        View="Week"
+                        QueryAppointments="OnSchedulerQueryAppointments">
+ </scheduler:SfScheduler>
+
+{% endhighlight %}
+{% highlight c# %}
+
+this.Scheduler.View = SchedulerView.Week;
+this.Scheduler.QueryAppointments += OnSchedulerQueryAppointments;
+
+private async void OnSchedulerQueryAppointments(object sender, SchedulerQueryAppointmentsEventArgs e)
+{
+    this.Scheduler.ShowBusyIndicator = true;
+    await Task.Delay(1500);
+    var appCollection = this.GenerateSchedulerAppointments(e.VisibleDates);
+    if (this.Scheduler.View != SchedulerView.Agenda)
+    {
+        this.Scheduler.AppointmentsSource = appCollection;
+    }
+    else
+    {
+        foreach (var app in appCollection)
+        {
+            ((ObservableCollection<SchedulerAppointment>)this.Scheduler.AppointmentsSource).Add(app);
+        }
+    }
+    this.Scheduler.ShowBusyIndicator = false;
+}
+
+private ObservableCollection<SchedulerAppointment> GenerateSchedulerAppointments(List<DateTime> visibleDates)
+{
+    var brush = new ObservableCollection<Brush>
+    {
+        new SolidColorBrush(Color.FromArgb("#FF8B1FA9")),
+        new SolidColorBrush(Color.FromArgb("#FFD20100")),
+        new SolidColorBrush(Color.FromArgb("#FFFC571D")),
+        new SolidColorBrush(Color.FromArgb("#FF36B37B")),
+        new SolidColorBrush(Color.FromArgb("#FF3D4FB5")),
+        new SolidColorBrush(Color.FromArgb("#FFE47C73")),
+        new SolidColorBrush(Color.FromArgb("#FF636363")),
+        new SolidColorBrush(Color.FromArgb("#FF85461E")),
+        new SolidColorBrush(Color.FromArgb("#FF0F8644")),
+        new SolidColorBrush(Color.FromArgb("#FF01A1EF"))
+    };
+
+    var subjectCollection = new ObservableCollection<string>
+    {
+        "Business Meeting",
+        "Conference",
+        "Medical check up",
+        "Performance Check",
+        "Consulting",
+        "Project Status Discussion",
+        "Client Meeting",
+        "General Meeting",
+        "Yoga Therapy",
+        "GoToMeeting",
+        "Plan Execution",
+        "Project Plan"
+    };
+
+    Random ran = new();
+    int daysCount = visibleDates.Count;
+    DateTime visibleStartDate = visibleDates.FirstOrDefault();
+    var appointments = new ObservableCollection<SchedulerAppointment>();
+    for (int i = 0; i < 25; i++)
+    {
+        var startTime = visibleStartDate.AddDays(ran.Next(0, daysCount + 1)).AddHours(ran.Next(9, 16));
+        appointments.Add(new SchedulerAppointment()
+        {
+            StartTime = startTime,
+            EndTime = startTime.AddHours(1),
+            Subject = subjectCollection[ran.Next(0, subjectCollection.Count)],
+            Background = brush[ran.Next(0, brush.Count)]
+        });
+    }
+
+    return appointments;
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+N> The `QueryAppointments` event will be raised if any one of the following action is performed,
```

Comment:
The `QueryAppointments` event will be triggered, if any one of the following actions is performed.